doc/sphinx/proof-engine/tactics.rst

```diff
@@ -909,196 +910,216 @@ quantification or an implication.
+      the source context, as long as the latter are also renamed by the same
+      tactic.
 
-.. exn:: No such hypothesis.
-.. exn:: @ident is already used.
+   .. exn:: No such hypothesis.
+      :undocumented:
+
+   .. exn:: @ident is already used.
+      :undocumented:
 
 .. tacn:: set (@ident := @term)
    :name: set
 
-   This replaces :n:`@term` by :n:`@ident` in the conclusion of the current goal
-   and adds the new definition :g:`ident := term` to the local context.
+   This replaces :token:`term` by :token:`ident` in the conclusion of the
+   current goal and adds the new definition :n:`@ident := @term` to the
+   local context.
 
-   If :n:`@term` has holes (i.e. subexpressions of the form “`_`”), the tactic
-   first checks that all subterms matching the pattern are compatible before
-   doing the replacement using the leftmost subterm matching the pattern.
+   If :token:`term` has holes (i.e. subexpressions of the form “`_`”), the
+   tactic first checks that all subterms matching the pattern are compatible
+   before doing the replacement using the leftmost subterm matching the
+   pattern.
 
-.. exn:: The variable @ident is already defined.
+   .. exn:: The variable @ident is already defined.
+      :undocumented:
 
-.. tacv:: set (@ident := @term ) in @goal_occurrences
+   .. tacv:: set (@ident := @term) in @goal_occurrences
 
-   This notation allows  specifying which occurrences of :n:`@term` have to be
-   substituted in the context. The :n:`in @goal_occurrences` clause is an
-   occurrence clause whose syntax and behavior are described in
-   :ref:`goal occurences <occurencessets>`.
+      This notation allows specifying which occurrences of :token:`term` have
+      to be substituted in the context. The :n:`in @goal_occurrences` clause
+      is an occurrence clause whose syntax and behavior are described in
+      :ref:`goal occurences <occurencessets>`.
 
-.. tacv:: set (@ident {+ @binder} := @term )
+   .. tacv:: set (@ident @binders := @term) {? in @goal_occurrences }
 
-   This is equivalent to :n:`set (@ident := funbinder {+ binder} => @term )`.
+      This is equivalent to :n:`set (@ident := fun @binders => @term) {? in @goal_occurrences }`.
 
-.. tacv:: set term
-   This behaves as :n:`set(@ident := @term)` but :n:`@ident` is generated by
-   Coq. This variant also supports an occurrence clause.
+   .. tacv:: set @term {? in @goal_occurrences }
 
-.. tacv:: set (@ident {+ @binder} := @term) in @goal_occurrences
-.. tacv:: set @term in @goal_occurrences
+      This behaves as :n:`set (@ident := @term) {? in @goal_occurrences }`
+      but :token:`ident` is generated by Coq.
 
-   These are the general forms that combine the previous possibilities.
+   .. tacv:: eset (@ident {? @binders } := @term) {? in @goal_occurrences }
+      :name: eset
 
-.. tacv:: eset (@ident {+ @binder} := @term ) in @goal_occurrences
-.. tacv:: eset @term in @goal_occurrences
-   :name: eset
+      .. eset @term {? in @goal_occurrences }
```

Comment:
@cpitclaudel I wanted to do the following:

```
.. tacv:: eset (@ident {? @binders } := @term) {? in @goal_occurrences }
          eset @term {? in @goal_occurrences }
   :name: eset
```

But this fails because it expects two indexes. Is there a workaround?

Reply:
Would something like `:name: eset; eset-short` work (that is, giving a second name)
I could also change the code, of course.

Reply:
Sure, but it would look weird to see `eset-short` in the tactic index. How complicated would it be to change the code? For instance to support a discarding pattern `:name: eset; _`

Reply:
Ah, true. Yes, that change would be easy.
But shouldn't we just separate 'name' and 'index text' entries?  Originally index entries and names were all auto-generated.  We switched to manually generated for some of them, but there's no reason to use the same for both (In the LaTeX doc we use to handwrite the indices).
Maybe we should have a separate discussion about this. In the meantime, you can change this:
```
    def add_target_and_index(self, name, _, signode):
        """Attach a link target to `signode` and an index entry for `name`."""
        if name:
            target = self._add_target(signode, name)
```
to this:
```
    def add_target_and_index(self, name, _, signode):
        """Attach a link target to `signode` and an index entry for `name`."""
        if name and not name.startswith("_"):
            target = self._add_target(signode, name)
```
and then any name prefixed with "_" should be omitted from the index.